### PR TITLE
merge with apoelstra/master

### PR DESCRIPTION
**Pull request by rphmeier**
includes a bump to newer libsecp

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -43,8 +43,8 @@ fn main() {
                .define("ENABLE_MODULE_RECOVERY", Some("1"));
 
     // secp256k1
-    base_config.file("depend/secp256k1/src/laxder_shim.c")
-               .file("depend/secp256k1/src/ext.c")
+    base_config.file("depend/secp256k1/contrib/lax_der_parsing.c")
+               .file("depend/secp256k1/src/secp256k1.c")
```
Comment (arkpar) [inline build.rs]: We still need ext.c I think unless master has a replacement for `secp256k1_ecdh_raw`

**Comment by arkpar** [comment]
Please test that parity tests pass with this version.

**Comment by rphmeier** [comment]
The build.rs needed some further tweaking as you suspected.

I have a travis build running here to test compatibility: https://github.com/rphmeier/parity/pull/1
and one with ethkey (which also uses it): https://github.com/rphmeier/ethkey/pull/1

**Comment by rphmeier** [comment]
they both look to have passed.

**Pull request merged**